### ISSUE TITLE
Engine Cleanup

### ISSUE THREAD

**Issue by rengaw83** [DMKEBUSINESSGMBH/typo3-mksearch#52]
The SearchEngineInterface defines a lot of useless methods like deleteIndex. 
The engines themselves implement the methods, most with useless copy'n'paste code.  
The engines have to be reconsidered and cleaned up.

**Comment by Baachi** [comment]
Yeah the engine interface is a mess. What i imagine would be the following:

We have a engine which is only responsible for all read access like searching or find a document. This could looks like this:

```php
interface EngineInterface
{
    function search(Query $query): iterabel;

    function find(string $extKey, int $uid): Document;
}
```
In opposite to the engine we could have an separate interface for all operations to manipulate the data.

```
interface IndexInterface
{
     function createDocument(Document $document): void;

     function deleteDocument(Document $document): void;

     function updateDocument(Document $document): void;

     function commit(): void;
}
```

Would you agree?

/cc @hannesbochmann @digedag
